### ISSUE TITLE
Automatically lowercase server_name

### ISSUE THREAD

**Issue by aaronraimist** [matrix-org/synapse#7728]
or complain if uppercase letters in server_name and tell user to change

**Comment by clokep** [comment]
Is there specific APIs you're thinking of here or anywhere that an MXID is handled by Synapse?

**Comment by richvdh** [comment]
I assume this refers to the config file setting.

**Comment by aaronraimist** [comment]
That’s right, this is talking about setting the server name in `homeserver.yaml`.

A user set their server name to have the first letter capitalized so the federation tester showed the matching server name error. They had to go back and erase their database and set the server name to be all lower case. It would be great if there was some warning that you set it improperly.

(I wish !github create would add a link to the context)

**Comment by clokep** [comment]
Makes sense. At the very least we should document it has to be all lower-case. Thanks for the additional info!
